src/views/virtualmachines/details/tabs/configuration/scripts/ScriptsTab.tsx

```diff
@@ -49,14 +50,15 @@ const ScriptsTab: FC<VirtualMachineScriptPageProps> = ({ obj: vm }) => {
   const { createModal } = useModal();
   const [authorizedSSHKeys, updateAuthorizedSSHKeys, loaded] = useKubevirtUserSettings('ssh');
   const secretName = useMemo(() => getVMSSHSecretName(vm), [vm]);
-  const isDynamicSSHInjection = useMemo(() => getIsDynamicSSHInjectionEnabled(vm), [vm]);
+  const isDynamicSSHInjectionTemplate = useMemo(() => getIsDynamicSSHInjectionEnabled(vm), [vm]);
+  const isDynamicSSHInjectionInstanceType = useDynamicSSHInjectionEnabledForVM(vm);
```

Comment:
both constants are used together wherever they are used, combine the getIsDynamicSSHInjectionEnabled function with the useDynamicSSHInjectionEnabledForVM hook.
Also useDynamicSSHInjection would be fine as a name for the hook if you want a shorter one